e2e/setup/vault.go

```diff
@@ -0,0 +1,63 @@
+package setup
+
+import (
+	"context"
+
+	"github.com/giantswarm/e2e-harness/v2/pkg/release"
+	"github.com/giantswarm/e2etemplates/pkg/chartvalues"
+	"github.com/giantswarm/microerror"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/azure-operator/v4/e2e/env"
+	"github.com/giantswarm/azure-operator/v4/e2e/key"
+)
+
+func installVault(ctx context.Context, config Config) error {
```

Comment:
could we install a public `vault` chart that gets configured how we want? That way we don't need the `chartvalues` dependency anymore, and we can get rid of the `registryPullSecret` env var.